### PR TITLE
Fix crash in D-down menu with World.dungeon_rewards_hinted off

### PR DESCRIPTION
**Pull request by fenhl**
There was [a report](https://discord.com/channels/274180765816848384/438698093354156032/1298744812668719125) about the game crashing when D-pad down is pressed on the inventory screen while `World.dungeon_rewards_hinted` is false. I have been unable to reproduce the crash on BizHawk or Project64 (the report was for Wii VC and RetroArch), but if @rrealmuto's analysis is correct, this should fix it.

### CODE REVIEW

**Comment by rrealmuto** [comment]
Did you try the specific set of settings that I mentioned? I reproduced in PJ64 no problem. I don't believe it is specific to my branch

**Comment by rrealmuto** [comment]
I will try merging this on my branch tonight though and see if it changes anything

**Comment by fenhl** [comment]
Ah, I'd missed that message. So this plando repros:

```json
{
    "settings": {
        "enhance_map_compass": true,
        "shuffle_mapcompass": "remove",
        "misc_hints": []
    }
}
```

And I can confirm this PR fixes it.

**Comment by rrealmuto** [comment]
🥳 idk if there are any other settings combinations that we might need to consider. Haven't actually looked at your changes yet

**Comment by fenhl** [comment]
The fix is on the asm level so we should be good.

**Comment by cjohnson57** [comment]
I read his explanation for why this was the issue, but for completeness and easy searchability could you summarize it here?

**Review by cjohnson57 — APPROVED** [review]

**Comment by fenhl** [comment]
[The patch populating the list of dungeon reward areas for the D-down menu](https://github.com/OoTRandomizer/OoT-Randomizer/blob/41cbf89232471bdee058f7ee4438ce2fdd3fd657/Patches.py#L2788-L2794) is skipped if `World.dungeon_rewards_hinted` is false (because the required information isn't computed), leaving [the default from `config.asm`](https://github.com/OoTRandomizer/OoT-Randomizer/blob/41cbf89232471bdee058f7ee4438ce2fdd3fd657/ASM/src/config.asm#L325-L327) in place. That default was all spaces (0x20) but the text rendering code expects null-terminated strings.
